src/pytest_plugins/forks/forks.py

```diff
@@ -52,6 +56,98 @@ def pytest_addoption(parser):
     )
 
 
+@dataclass(kw_only=True)
+class ForkCovariantParameter:
+    """
+    Value list for a fork covariant parameter in a given fork.
+    """
+
+    name: str
+    values: List[Any]
+
+
+@dataclass(kw_only=True)
+class ForkParametrizer:
+    """
+    A parametrizer for a test case that is parametrized by the fork.
+    """
+
+    fork: Fork
+    mark: pytest.MarkDecorator | None = None
+    fork_covariant_parameters: List[ForkCovariantParameter] = field(default_factory=list)
+
+    def get_parameter_names(self) -> List[str]:
+        """
+        Return the parameter names for the test case.
+        """
+        return ["fork"] + [p.name for p in self.fork_covariant_parameters]
+
+    def get_parameter_values(self) -> List[Any]:
+        """
+        Return the parameter values for the test case.
+        """
+        return [
+            pytest.param(*params, marks=[self.mark] if self.mark else [])
+            for params in itertools.product(
+                [self.fork],
+                *[p.values for p in self.fork_covariant_parameters],
+            )
+        ]
+
+
+@dataclass(kw_only=True)
+class CovariantDescriptor:
+    """
+    A descriptor for a parameter that is covariant with the fork:
+    the parametrized values change depending on the fork.
+    """
+
+    marker_name: str
+    description: str
+    fork_attribute_name: str
+    parameter_name: str
+
+    def check_enabled(self, metafunc: Metafunc) -> bool:
+        """
+        Check if the marker is enabled for the given test function.
+        """
+        m = metafunc.definition.iter_markers(self.marker_name)
+        return m is not None and len(list(m)) > 0
+
+    def add_values(self, metafunc: Metafunc, fork_parametrizer: ForkParametrizer) -> None:
+        """
+        Add the values for the covariant parameter to the parametrizer.
+        """
+        if not self.check_enabled(metafunc=metafunc):
+            return
+        fork = fork_parametrizer.fork
+        fork_attribute: Callable[[int, int], List[Any]] = getattr(fork, self.fork_attribute_name)
+        values = fork_attribute(0, 0)
+        assert isinstance(values, list)
+        assert len(values) > 0
+        fork_parametrizer.fork_covariant_parameters.append(
+            ForkCovariantParameter(name=self.parameter_name, values=values)
+        )
```

Comment:
How about this small change to help readability?
```suggestion
        get_fork_covariant_values: Callable[[int, int], List[Any]] = getattr(fork, self.fork_attribute_name)
        values = get_fork_covariant_values(block_number=0, timestamp=0)
        assert isinstance(values, list)
        assert len(values) > 0
        fork_parametrizer.fork_covariant_parameters.append(
            ForkCovariantParameter(name=self.parameter_name, values=values)
        )
```